content/academy/expert_scraping_with_apify/solutions/rotating_proxies.md

```diff
@@ -0,0 +1,115 @@
+
+```JavaScript
+const crawler = new Apify.CheerioCrawler({
+    requestList,
+    requestQueue,
+    proxyConfiguration,
+    useSessionPool: true,
+    sessionPoolOptions: {
+        persistStateKey: 'AMAZON-SESSIONS',
+        sessionOptions: {
+            maxUsageCount: 5,
+            maxErrorScore: 1,
+        },
+    },
+    maxConcurrency: 50,
+    // ...
+});
+```
+
+And that's it! We've successfully configured the session pool to match the task's requirements.
+
+## [](#limiting-proxy-location) Limiting proxy location
+
+The final requirement was to only use proxies from the US. Back in our **ProxyConfiguration**, we just need to add the **countryCode** key and set it to **US**:
+
+```JavaScript
+const proxyConfiguration = await Apify.createProxyConfiguration({
+    groups: ['RESIDENTIAL'],
+    countryCode: 'US',
+});
+```
+
+## [](#quiz-answers) Quiz answers
+
+**Q: What are the different types of proxies that Apify proxy offers? What are the main differences between them?**
+
+**A:** Datacenter, residential, and Google SERP proxies with sub-groups. Datacenter proxies are fast and cheap, but have a higher chance of being blocked on certain sites in comparison to residential proxies, which are IP addresses located in homes and offices around the world. Google SERP proxies are specifically for Google.
+
+**Q: Which proxy groups do users get with the proxy trial? How long does the trial last?**
+
+**A:** Users can get a 30 day trial to Apify Proxy and will get access to the **GOOGLE_SERP** and **RESIDENTIAL** groups.
+
+**Q: How can you prevent an error from occurring if one of the proxy groups that a user has is removed? What are the best practices for these scenarios?**
+
+**A:** By making the proxy for the scraper to use be configurable by the user through the actor's input. That way, they can easily switch proxies if the actor stop working due to proxy-related issues.
+
+**Q: Does it make sense to rotate proxies when you are logged into a website?**
+
+**A:** No, because most websites tie an IP address to a session. If you start making requests with cookies used with a different IP address, the website might see it as unusual activity and either block the scraper or automatically log out.
+
+**Q: Construct a proxy URL that will select proxies only from the US.**
+
+**A:** `http://country-US:<proxy_password>@proxy.apify.com:8000`
+
+**Q: What do you need to do to rotate a proxy (one proxy usually has one IP)? How does this differ for CheerioCrawler and PuppeteerCrawler?**
+
+**A:** Simply making a new request with the proxy endpoint above will automatically rotate it. Sessions can also be used to automatically do this. While proxy rotation is fairly straightforward for Cheerio, it's more complex in Puppeteer, as you have to retire the browser each time a new proxy is rotated in.
```

Comment:
Add: If you use SessionPool (which is used by default), the browser is retired once a session is retired. You can manually retire via `session.retire()`